R/baggr_compare.R

```diff
@@ -57,33 +87,41 @@
 
 baggr_compare <- function(...,
                           what    = "pooling",
-                          compare = "groups",
-                          style   = "areas",
-                          arrange = "single") {
+                          compare = "groups") {
   l <- list(...)
   if(length(l) == 0)
     stop("Must provide baggr models or model specification.")
   if(all(unlist(lapply(l, class)) == "baggr")) {
-    return_models_flag <- 0
+    # return_models_flag <- 0
     if(is.null(names(l)))
       names(l) <- paste("Model", 1:length(l))
     models <- l
   } else {
-    return_models_flag <- 1
+    # return_models_flag <- 1
     if(what == "pooling"){
       if("pooling" %in% names(l))
         stop("Can't run the model comparison with pooling setting",
              "already set to a particular value.")
       models <- lapply(list("none", "partial", "full"), function(pool){
-        try(do.call(baggr, c(l, "pooling" = pool)))
+        # message to display progress
+        message(paste0("Sampling for model with pooling set to ", pool))
+
+        # suppress baggr/rstan output
+        make_silent(model <- do.call(baggr, c(l, "pooling" = pool)))
+
+        # return model
+        model
       })
       names(models) <- c("none", "partial", "full")
     } else if(what == "prior") {
       if("ppd" %in% names(l))
         stop("Can't run the model comparison with ppd setting",
              "already set to a particular value.")
       models <- lapply(list(TRUE, FALSE), function(ppdv){
-        try(do.call(baggr, c(l, "ppd" = ppdv)))
+        check_which <- ifelse(ppdv, "just the prior", "prior and full data")
+        message(paste0("Sampling for model with ", check_which, "."))
+        make_silent(model <- do.call(baggr, c(l, "ppd" = ppdv)))
```

Comment:
why not `try` anymore?

Reply:
@wwiecek Per our discussion yesterday I'm putting together a safer solution. I think it would be helpful to have the ability to silence the prepare_prior messages, so I'm creating a pull request that adds that option. If that's not an approach you'd like use feel free to reject. Pretty minimal impact on the project though.

Reply:
I think for that you would want an argument `silent` in call to `baggr()`, which is then passed to `prepare_prior()`, `convert_inputs()`, the function that detects input type etc...? In any case, yes, that will be a small but useful (separate) PR.

Reply:
I added it into this one, but I can separate it into a separate PR.